File: packages/types/src/work-item-type.ts
/**
 * Copyright (c) 2023-present Plane Software, Inc. and contributors
 * SPDX-License-Identifier: AGPL-3.0-only
 * See the LICENSE file for details.
 */

export type IWorkItemType = {
  id: string;
  name: string;
  description: string;
  logo_props: Record<string, unknown>;
  is_default: boolean;
  is_active: boolean;
  is_epic: boolean;
  created_at: string;
};

export type IWorkItemTypeFormData = {
  name: string;
  description?: string;
  logo_props?: Record<string, unknown>;
  is_default?: boolean;
  is_active?: boolean;
};

File: apps/api/plane/app/serializers/work_item_type.py
# Copyright (c) 2023-present Plane Software, Inc. and contributors
# SPDX-License-Identifier: AGPL-3.0-only
# See the LICENSE file for details.

from .base import BaseSerializer
from plane.db.models import IssueType


class WorkItemTypeSerializer(BaseSerializer):
    class Meta:
        model = IssueType
        fields = [
            "id",
            "name",
            "description",
            "logo_props",
            "is_default",
            "is_active",
            "is_epic",
            "created_at",
        ]
        read_only_fields = ["id", "created_at"]
